### PR TITLE
feat: Implement Loss Functions

### PR DESCRIPTION
**Pull request by soumik12345**
Fixes https://github.com/soumik12345/pynet/issues/2
Fixes https://github.com/soumik12345/pynet/issues/3

- [x] Implement workflow for code-check
- [x] Implement workflow for executing unit tests
- [x] Refactor `pynet.losses` module
- [x] Implement `SSIMLoss`
- [x] Implement `MultiScaleSSIMLoss`
- [x] Implement `PerceptualLoss`
- [X] Write unit tests for `pynet.losses` module

### CODE REVIEW

File: pynet/losses.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import tensorflow as tf
+from keras.utils import losses_utils
```
Comment (sanatsingh) [inline pynet/losses.py:2]: Unused import can be removed

File: pynet/tests/test_losses.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class LossTester(unittest.TestCase):
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+        self.ssim_loss = SSIMLoss()
+        self.multi_scale_ssim_loss = MultiScaleSSIMLoss()
+        self.perceptual_loss = PerceptualLoss(rescale_inputs=False)
+        self.sample_y = tf.ones((1, 224, 224, 3))
+
+    def test_ssim_loss(self):
+        self.assertEqual(self.ssim_loss(self.sample_y, self.sample_y).numpy(), 1.0)
```
Comment (sanatsingh) [inline pynet/tests/test_losses.py:17]: Formatting mismatch

File: pynet/tests/test_pynet.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import unittest
+
+import tensorflow as tf
+
+from pynet.model.modules import level_0, level_1, level_2, level_3, level_4, level_5
+from pynet.model.pynet import PyNet
+
+
+class ModelTester(unittest.TestCase):
```
Comment (sanatsingh) [inline pynet/tests/test_pynet.py:9]: Let's make this similar to LossTester class, more efficient and readable.
E.g.
```suggestion
class ModelTester(unittest.TestCase):
    def __init__(self, methodName: str = "runTest") -> None:
        super().__init__(methodName)
        l4_pool = tf.ones((1, 14, 14, 256))
        l4_out_1 = tf.ones((1,28,28,256))
        l5_pass_a = tf.ones((1,28,28,256))
        l5_pass_b = tf.ones((1,28,28,256))
        ...
```

File: pynet/losses.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        if self.rescale_inputs:
+            y_true = y_true * 255.0
+            y_pred = y_pred * 255.0
+        y_true_features = self.vgg_feature_model(
+            keras.applications.vgg19.preprocess_input(y_true)
+        )
+        y_pred_features = self.vgg_feature_model(
+            keras.applications.vgg19.preprocess_input(y_pred)
+        )
+        return tf.reduce_mean(self.mean_squared_error(y_true, y_pred))
```
[the remark this answers is not in the document]
Comment (sanatsingh) [reply pynet/losses.py:49]: The reference is using reduction = 'None' for tf.losses.MeanSquaredError, hence they are applying mean later

**Review by sanatsingh — COMMENTED** [review]
Everything looks good, nice work. Few changes required and then we are good to go

File: pynet/tests/test_losses.py (hunk 2 above)
Comment (soumik12345) [reply pynet/tests/test_losses.py:17]: @sanatsingh 
Can you please elaborate this?
